### PR TITLE
Update timestamp columns in measuremets on PostgreSQL

### PR DESCRIPTION
**Pull request by eitoball**
See Safecast/safecastapi#354

### CODE REVIEW

File: db/migrate/20170201124248_add_defaults_to_timestamps_in_measurements.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class AddDefaultsToTimestampsInMeasurements < ActiveRecord::Migration[5.0]
+  def up
+    change_table :measurements do |t|
+      %i(created_at updated_at).each do |col|
+        t.change col, :datetime, null: true, default: -> { 'now()' }
+      end
+    end
+
+    execute <<-__END__
```
Comment (matschaffer) [inline db/migrate/20170201124248_add_defaults_to_timestamps_in_measurements.rb]: Can probably indent this when using `<<-`

I also like to use `SQL` as the delimiter. That way IntelliJ (I think some others too) can syntax highlight within the heredoc.

File: application.rb [hunk 2]
```diff
@@ -19,3 +19,7 @@
 ].each do |pattern|
   Dir.glob(Config.root.join(*pattern)).each { |file| require file }
 end
+
+::ActiveRecord::Base.schema_format = :sql
+# XXX: Update created_at and updated_at in PostgreSQL
+::Measurement.record_timestamps = false
```
Comment (matschaffer) [inline application.rb]: Seems like this should go into the measurement class directly. Or at least the same file.

Unclear on what the XXX signifies.

File: db/structure.sql [hunk 3]
```diff
@@ -0,0 +1,148 @@
+--
+-- PostgreSQL database dump
```
Comment (matschaffer) [inline db/structure.sql]: 👍  for sql structure. We should probably remove the schema.rb in this PR though.

File: application.rb (hunk 2 above)
Comment (eitoball) [inline application.rb]: Will move it.

> Unclear on what the XXX signifies.

I am not sure I use it in right way, but

```
XXX - warn other programmers of problematic or misguiding code
```
https://en.wikipedia.org/wiki/Comment_(computer_programming)

Comment (matschaffer) [inline application.rb]: Huh. TIL

File: db/migrate/20170201124248_add_defaults_to_timestamps_in_measurements.rb (hunk 1 above)
Comment (eitoball) [inline db/migrate/20170201124248_add_defaults_to_timestamps_in_measurements.rb]: Indented.

> That way IntelliJ (I think some others too) can syntax highlight within the heredoc.

Interesting. I will check with RubyMine for syntax highlight.

File: db/migrate/20170201124248_add_defaults_to_timestamps_in_measurements.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
+class AddDefaultsToTimestampsInMeasurements < ActiveRecord::Migration[5.0]
+  def up
+    change_table :measurements do |t|
+      %i(created_at updated_at).each do |col|
+        t.change col, :datetime, null: true, default: -> { 'now()' }
+      end
+    end
+
+    execute <<-SQL
+      CREATE OR REPLACE FUNCTION update_updated_at_column()
```
Comment (matschaffer) [inline db/migrate/20170201124248_add_defaults_to_timestamps_in_measurements.rb]: Looks like even github does that highlighting trick :)

File: db/structure.sql (hunk 3 above)
Comment (matschaffer) [inline db/structure.sql]: Let's rebase this now that #8 is merged and ensure the device/device_seq story is sane.

Comment (eitoball) [inline db/structure.sql]: Rebased

File: db/structure.sql [hunk 5]
```diff
@@ -232,7 +253,8 @@ INSERT INTO schema_migrations (version) VALUES
 ('20170126020054'),
 ('20170126024027'),
 ('20170130081839'),
-('20170202014348'),
+('20170201124248'),
+('20170202014348');
```
Comment (matschaffer) [inline db/structure.sql]: Looks like a bad merge here with the semicolon mid-line

**Comment by eitoball** [comment]
> Might be best to load master's structure.sql then re-run the migrations to get the right final structure exported.

Done. Thank you for catching.

**Pull request merged**